packages/api-derive/src/staking/info.ts

```diff
@@ -2,42 +2,110 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
-import { DerivedStaking } from '../types';
+import { ApiInterface$Rx } from '@polkadot/api/types';
+import { AccountId, BlockNumber, Exposure, Option, StakingLedger,StructAny, ValidatorPrefs, UnlockChunk } from '@polkadot/types';
+import { DerivedStaking, DerivedUnlockingMap } from '../types';
 
-import { Observable, of } from 'rxjs';
+import BN from 'bn.js';
+import { combineLatest, Observable, of } from 'rxjs';
 import { map, switchMap } from 'rxjs/operators';
-import { ApiInterface$Rx } from '@polkadot/api/types';
-import { AccountId, Exposure, Option, StakingLedger, StructAny, ValidatorPrefs } from '@polkadot/types';
 
+import { bestNumber } from '../chain/bestNumber';
 import { drr } from '../util/drr';
+import { eraLength } from '../session/eraLength';
+import { isUndefined } from '@polkadot/util';
+
+function calculateUnlocking (stakingLedger: StakingLedger | undefined, eraLength: BN, bestNumber: BlockNumber) {
+  if (isUndefined(stakingLedger)) return undefined;
+
+  // select the Unlockchunks that can't be redeemed yet.
+  const unlockingChunks = stakingLedger.unlocking.filter((chunk) => remainingBlocks(chunk.era, eraLength, bestNumber).gtn(0));
+
+  if (unlockingChunks.length) {
+    // group the Unlockchunks that have the same era and sum their values
+    const groupedResult = groupByEra(unlockingChunks);
+    const results: DerivedUnlockingMap = [];
+
+    Object.keys(groupedResult).map((eraString) => (
+      results.push({ value: groupedResult[eraString], remainingBlocks: remainingBlocks(new BlockNumber(eraString), eraLength, bestNumber) })
```

Comment:
One of the reasons for the return type on the function, we are not actually using `eraLength` or `bestNumber`

Should just be, with readability enhancements -

```js
results.push({
  remainingBlocks: remainingBlocks(new BlockNumber(eraString)),
  value: groupedResult[eraString]
})
```